### ISSUE TITLE
Rails/LexicallyScopedActionFilter: false positive with ActiveSupport::Concern

### ISSUE THREAD

**Issue by stripedpumpkin** [bbatsov/rubocop#5605]
When defining a concern in app/controllers/food_controlling.rb:

```ruby
# frozen_string_literal: true

module FoodControlling
  extend ActiveSupport::Concern

  included do
    before_action :serve_the_food, only: [:check_fresh]
  end

  private

  def serve_the_food
    head :ok
  end

  def check_fresh
    raise 'NotFresh' unless food.fresh?
  end
end
```

rubocops output a false positive for Rails/LexicallyScopedActionFilter.

--------

## RuboCop version

```
$ rubocop -V
bundle exec rubocop -V
0.52.1 (using Parser 2.5.0.2, running on ruby 2.4.1 x86_64-linux)
```

## Expected behavior
```bash
 bundle exec rubocop app/controllers/food_controlling.rb -C false
=> no offense
```
## Actual behavior

```bash
 bundle exec rubocop app/controllers/food_controlling.rb -C false
=> app/controllers/food_controlling.rb:7:5: C: Rails/LexicallyScopedActionFilter: check_fresh is not explicitly defined on the controller.
    before_action :serve_the_food, only: [:check_fresh]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Trace

```bash
[68, 77] in /home/pumpkin/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.52.1/lib/rubocop/cop/rails/lexically_scoped_action_filter.rb
   68:         def on_send(node)
   69:           byebug
   70:           methods_node = only_or_except_filter_methods(node)
   71:           return unless methods_node
   72: 
=> 73:           defined_methods = node.parent.each_child_node(:def).map(&:method_name)
   74:           methods = array_values(methods_node).reject do |method|
   75:             defined_methods.include?(method)
   76:           end
   77: 
(byebug) node
s(:send, nil, :before_action,
  s(:sym, :serve_the_food),
  s(:hash,
    s(:pair,
      s(:sym, :only),
      s(:array,
        s(:sym, :check_fresh)))))
(byebug) node.parent
s(:block,
  s(:send, nil, :included),
  s(:args),
  s(:send, nil, :before_action,
    s(:sym, :serve_the_food),
    s(:hash,
      s(:pair,
        s(:sym, :only),
        s(:array,
          s(:sym, :check_fresh))))))
(byebug) node.parent.each_child_node(:def).map(&:method_name)
[]
(byebug) next

[69, 78] in /home/pumpkin/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rubocop-0.52.1/lib/rubocop/cop/rails/lexically_scoped_action_filter.rb
   69:           byebug
   70:           methods_node = only_or_except_filter_methods(node)
   71:           return unless methods_node
   72: 
   73:           defined_methods = node.parent.each_child_node(:def).map(&:method_name)
=> 74:           methods = array_values(methods_node).reject do |method|
   75:             defined_methods.include?(method)
   76:           end
   77: 
   78:           add_offense(node, message: message(methods)) unless methods.empty?
(byebug) methods_node
s(:array,
  s(:sym, :check_fresh))
(byebug) node.parent.parent
s(:begin,
  s(:send, nil, :extend,
    s(:const,
      s(:const, nil, :ActiveSupport), :Concern)),
  s(:block,
    s(:send, nil, :included),
    s(:args),
    s(:send, nil, :before_action,
      s(:sym, :serve_the_food),
      s(:hash,
        s(:pair,
          s(:sym, :only),
          s(:array,
            s(:sym, :check_fresh)))))),
  s(:send, nil, :private),
  s(:def, :serve_the_food,
    s(:args),
    s(:send, nil, :head,
      s(:sym, :ok))),
  s(:def, :check_fresh,
    s(:args),
    s(:if,
      s(:send,
        s(:send, nil, :food), :fresh?), nil,
      s(:send, nil, :raise,
        s(:str, "NotFresh")))))
(byebug) node.parent.parent.each_child_node(:def).map(&:method_name)
[:serve_the_food, :check_fresh]
```

**Comment by wata727** [comment]
This false positive already fixed in https://github.com/bbatsov/rubocop/pull/5457
Please check current [master](https://github.com/bbatsov/rubocop/tree/7868d5229ade75ad3e1ba80e792ea51352188880) branch.

```
$ rubocop _0.52.1_ --only Rails/LexicallyScopedActionFilter app/controllers/food_controlling.rb
Inspecting 1 file
C

Offenses:

app/controllers/food_controlling.rb:7:5: C: Rails/LexicallyScopedActionFilter: check_fresh is not explicitly defined on the controller.
    before_action :serve_the_food, only: [:check_fresh]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

```
$ bin/rubocop --only Rails/LexicallyScopedActionFilter app/controllers/food_controlling.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by stripedpumpkin** [comment]
@wata727 Good new. Thank you. Closing here.
